src/orchestration/Orchestration.ts

```diff
@@ -276,6 +278,14 @@ export class Orchestration {
         this.dispatchManager.setAwsCredentials(credentials);
     }
 
+    /**
+     * Set custom session attributes to add them to all event metadata.
+     * @param payload object containing custom attribute data in the form of key, value pairs
+     */
+    public setCustomAttributes(payload: { [key: string]: any }): void {
```

Comment:
todo: We'll need to brainstorm some names for this API. E.g.,
- `addAttribute`
- `setAttribute`
- `setSessionAttribute`
- `setSessionMetadata`

Reply:
issue: Could we replace the `any` type with `string | boolean | number`? This would help communicate the expected attribute types to callers.

Reply:
Updated to use `setSessionAttribute`. Replaced `any` with `string | boolean | number`